reconcile/ldap_groups/integration.py

```diff
@@ -147,38 +147,47 @@ def get_roles(self, query_func: Callable) -> list[RoleV1]:
         data = roles_query(query_func, variables={})
         roles = [role for role in data.roles or []]
         if duplicates := find_duplicates(
-            role.ldap_group for role in roles if role.ldap_group
+            role.ldap_group.name for role in roles if role.ldap_group
         ):
             for dup in duplicates:
                 logging.error(f"{dup} is already in use by another role.")
             raise ValueError("Duplicate ldapGroup value(s) found.")
         return roles
 
     def get_desired_groups_for_roles(
-        self, roles: Iterable[RoleV1], owners: Iterable[Entity], contact_list: str
+        self,
+        roles: Iterable[RoleV1],
+        default_owners: list[Entity],
+        contact_list: str,
     ) -> list[Group]:
         """Return the desired rover groups for the given roles."""
-        return [
-            Group(
-                name=role.ldap_group,
-                description="Persisted App-Interface role. Managed by qontract-reconcile",
-                display_name=f"{role.ldap_group} (App-Interface))",
-                members=[
-                    Entity(type=EntityType.USER, id=user.org_username)
-                    for user in role.users
-                ],
-                # only owners can modify the group (e.g. add/remove members)
-                owners=owners,
-                contact_list=contact_list,
+        groups = []
+        for role in roles:
+            if not role.ldap_group:
+                continue
+            members = [
+                Entity(type=EntityType.USER, id=user.org_username)
+                for user in role.users
+            ]
+            groups.append(
+                Group(
+                    name=role.ldap_group.name,
+                    description="Persisted App-Interface role. Managed by qontract-reconcile",
+                    notes=role.ldap_group.notes,
+                    display_name=f"{role.ldap_group.name} (App-Interface))",
+                    members=members,
+                    owners=default_owners
+                    if not role.ldap_group.members_are_owners
+                    else default_owners + members,
```

Comment:
I dont think its a problem, but can it be an issue if a `member` is already part of the `default_owners` list, i.e., a `member` entity appears twice in the `owners` list?

Reply:
The `default_owners` are just our service accounts; therefore, it isn't a problem at all.